### DISCUSSION TITLE
Excitation signal distortion at wide frequency ranges?

### DISCUSSION THREAD

**Discussion by 0xCoto** [thliebig/openEMS-Project#23] [General]
It seems that perhaps at wide frequency ranges (centered around 0 Hz to give a single pulse with a shorter excitation length), the excitation signal gets distorted?

```python
f0 = 0 # 0 Hz
fc = 12000000000 # 12 GHz

FDTD = openEMS(NrTS=1e9, EndCriteria=float(end_criteria), TimeStepMethod=1)
FDTD.SetGaussExcite(f0, fc)
```
Expected excitation signal (red):
![image](https://user-images.githubusercontent.com/25392776/211201197-8265bfe3-ec7a-478c-8a8e-219faa65447d.png)

Excitation signal openEMS gives (blue):
![image](https://user-images.githubusercontent.com/25392776/211201219-86f2ff81-a46b-47aa-a57d-26060c8e513e.png)

I haven't seen such issues with narrow frequency ranges (e.g. 0-5 GHz) but I can't quite figure what could cause such distortion? The 
C++ source code doesn't seem to indicate anything particularly suspicious... Any ideas?

**Comment by thliebig** [comment]
The question is what is the blue curve??? And if this is purely "et" than I would say that this is indeed impossible... If it is anything else... why not?

**Reply by 0xCoto** [reply]
The blue curve is the incident voltage, but why does it give these ripples? This is how I'm plotting it:
```py
# Post-processing and plotting
f = np.linspace(f_min, f_max, 1001) #f_min is 0 and f_max = 12e9
port.CalcPort(path, f)

inc_t = port.ut_inc
ref_t = port.ut_ref

# Get timestep
file = open(path+"/port_ut_1")
ts = float(file.readlines()[5].split()[0])
timesteps = np.linspace(0, 1e9*ts*len(ref_t), len(ref_t))

plot(timesteps, inc_t) # blue curve (incident)
plot(timesteps, ref_t) # orange curve (reflection)
show()
```
Indeed, in theory any signal could be applied, but I don't understand why the Gaussian excitation (which uses a specific equation) would give these ripples at wide frequency ranges...

**Reply by thliebig** [reply]
That is a typical sign of mismatch and reflections out of and back into and back out of again... very much normal...

Only if you look at "et" you see the real "excitation" (Gaussian) signal. The incoming voltage is the excitation and already some reflections that have reflected again on the port and thus out it again...

**Reply by 0xCoto** [reply]
Ah ok. Surprisingly enough, I have never noticed this effect (the incident signal being affected, even for weak and strong reflections). For example, this dipole (0-2 GHz excitation, antenna resonant at 1 GHz) doesn't seem to have its blue curve affected at all, even though the reflection contains mismatches at various frequencies (other than 1 GHz):
![image](https://user-images.githubusercontent.com/25392776/211205942-c1860938-a6e4-4694-b380-715a06b8f789.png)

Not sure why this horn antenna seems to behave this way...

Anyway, looks like plotting et gives the curve I expected:
![image](https://user-images.githubusercontent.com/25392776/211205505-81b6f178-24bb-40ca-a41d-fa1c513b9dc1.png)

I guess the other software (CST) works the same way as openEMS (i.e. displaying the incident voltage), but for whatever reason the reflection is different so the effect isn't as visible? Or are they probably displaying et as the red curve from the looks of it?

**Reply by thliebig** [reply]
Well the curves up there look very much different... 

Usually you get additional reflections from the port itself if it is not matched to the line/waveguide it is directly connected too. So not sure why this happens in your setup...

Edit: I always answer in the wrong comment section :|

**Reply by KJ7LNW** [reply]
What does it mean to have `f0 = 0`?  Doesn't that mean the bandwidth will be `f0 +/- fc`?

If so, then what does a _negative frequency_ mean?
